lib/reek/context/module_context.rb

```diff
@@ -7,6 +9,19 @@ module Context
     # A context wrapper for any module found in a syntax tree.
     #
     class ModuleContext < CodeContext
+      include Enumerable
```

Comment:
I think this will lead to confusing code elsewhere, and the fact that ModuleContext doesn't define `#each` itself is also confusing. I think we can just do `each.select` or `each.grep` below. I'd include Enumerable in ClassContext if it didn't also define other kinds of iterators.